engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -0,0 +1,312 @@
+  case object WorkflowExecutionAbortedResponse extends WorkflowExecutionActorResponse
+  final case class WorkflowExecutionFailedResponse(reasons: Seq[Throwable]) extends WorkflowExecutionActorResponse
+
+  /**
+    * Internal control flow messages
+    */
+  private case class JobInitializationFailed(throwable: Throwable)
+  private case class ScatterCollectionFailedResponse(collectorKey: CollectorKey, throwable: Throwable)
+  private case class ScatterCollectionSucceededResponse(collectorKey: CollectorKey, outputs: CallOutputs)
+
+  /**
+    * Internal ADTs
+    */
+  case class ScatterKey(scope: Scatter) extends JobKey {
+    override val index = None // When scatters are nested, this might become Some(_)
+    override val attempt = 1
+    override val tag = scope.unqualifiedName
+
+    /**
+      * Creates a sub-ExecutionStore with Starting entries for each of the scoped children.
+      *
+      * @param count Number of ways to scatter the children.
+      * @return ExecutionStore of scattered children.
+      */
+    def populate(count: Int): Map[JobKey, ExecutionStatus.Value] = {
+      val keys = this.scope.children flatMap { explode(_, count) }
+      keys map { _ -> ExecutionStatus.NotStarted } toMap
+    }
+
+    private def explode(scope: Scope, count: Int): Seq[JobKey] = {
+      scope match {
+        case call: Call =>
+          val shards = (0 until count) map { i => BackendJobDescriptorKey(call, Option(i), 1) }
+          shards :+ CollectorKey(call)
+        case scatter: Scatter =>
+          throw new UnsupportedOperationException("Nested Scatters are not supported (yet).")
+        case e =>
+          throw new UnsupportedOperationException(s"Scope ${e.getClass.getName} is not supported.")
+      }
+    }
+  }
+
+  // Represents a scatter collection for a call in the execution store
+  case class CollectorKey(scope: Call) extends JobKey {
+    override val index = None
+    override val attempt = 1
+    override val tag = s"Collector-${scope.unqualifiedName}"
+  }
+
+  case class WorkflowExecutionException(override val throwables: List[Throwable]) extends ThrowableAggregation {
+    override val exceptionContext = s"WorkflowExecutionActor"
+  }
+
+  def props(workflowId: WorkflowId, workflowDescriptor: EngineWorkflowDescriptor): Props = Props(WorkflowExecutionActor(workflowId, workflowDescriptor))
+}
+
+final case class WorkflowExecutionActor(workflowId: WorkflowId, workflowDescriptor: EngineWorkflowDescriptor) extends LoggingFSM[WorkflowExecutionActorState, WorkflowExecutionActorData] {
+
+  import WorkflowExecutionActor._
+  import lenthall.config.ScalaConfig._
+
+  val tag = self.path.name
+  private lazy val DefaultMaxRetriesFallbackValue = 10
+
+  // TODO: We should probably create a trait which loads all the configuration (once per application), and let classes mix it in
+  // to avoid doing ConfigFactory.load() at multiple places
```

Comment:
`ConfigFactory.load` is cached anyway, so we don't need to create a separate trait for this